presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1463,6 +1469,74 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeString(SqlBaseParser.UnicodeStringLiteralContext context)
+    {
+        char escape;
+        if (context.UESCAPE() != null) {
+            String escapeString = unquote(context.STRING().getText());
+            check(!escapeString.isEmpty(), "Empty Unicode escape character", context);
+            check(escapeString.length() == 1, "Invalid Unicode escape character: " + escapeString, context);
+            escape = escapeString.charAt(0);
+            check(isValidUnicodeEscape(escape), "Invalid Unicode escape character: " + escapeString, context);
+        }
+        else {
+            escape = '\\';
+        }
+
+        String rawContent = unquote(context.UNICODE_STRING().getText().substring(2));
+        StringBuilder builder = new StringBuilder();
+        int beginCodeIndex = 0;
+        int charactersNeeded = 0;
+        for (int i = 0; i < rawContent.length(); i++) {
+            if (rawContent.charAt(i) == escape) {
+                check(charactersNeeded == 0 && i < rawContent.length() - 1, "Incomplete escaped character: " + rawContent.substring(beginCodeIndex, i + 1), context);
```

Comment:
`Incomplete escape sequence:`

You are not displaying an escaped character here. You are showing the escape sequence.